### DISCUSSION TITLE
Problem with solving instance mps files

### DISCUSSION THREAD

**Discussion by cwfparsonson** [ds4dm/ml4co-competition#26] [Q&A] [answered]
Hi,

I'm having an issue using the scip solver to solve the .mps instances and I just wanted to sanity check if you'd expect my code to work or if perhaps there's been an error in the way I've downloaded and decompressed the .mps files:

```python
from ecole.scip import Model

path = '../milp/datasets/instances/1_item_placement/train/item_placement_1.mps'
model = Model.from_file(path)
m = model.as_pyscipopt()
m.optimize()
```

This code starts running but doesn't seem to stop. I've left it for ~30 minutes and it does not finish solving (but no exceptions are being raised). On the other hand, using the `mip` package to load and solve the same problem seems to work in <5 minutes, which suggests the .mps file is fine:

```python
import mip

path = '../milp/datasets/instances/1_item_placement/train/item_placement_1.mps'
instance = mip.Model()
instance.read(path)

status = instance.optimize(max_seconds=300)

if status == mip.OptimizationStatus.OPTIMAL:
    print('optimal solution cost {} found'.format(instance.objective_value))
elif status == mip.OptimizationStatus.FEASIBLE:
    print('sol.cost {} found, best possible: {}'.format(instance.objective_value, instance.objective_bound))
elif status == mip.OptimizationStatus.NO_SOLUTION_FOUND:
    print('no feasible solution found, lower bound is: {}'.format(instance.objective_bound))
```
```ansi
sol.cost 60.796965159000024 found, best possible: 28.922094929823203
```

Is my above implementation of `m.optimize()` correct? Would you expect the scip solver to solve this instance in <30 minutes?

**Comment by cwfparsonson** [comment, marked as answer]
Hi,

Sorry just realised that the reason this 'wasn't solving' was because I forgot to set the maximum time limit for the scip solver. The below works:

```python
from ecole.scip import Model

path = '../milp/datasets/instances/1_item_placement/train/item_placement_1.mps'
model = Model.from_file(path)
scip_params = {'limits/time': 300}
model.set_params(scip_params)
m = model.as_pyscipopt()
m.optimize()
```

**Reply by gasse** [reply]
Great :) No worries, don't hesitate to ask questions !
